src/main/scala/beam/utils/scenario/generic/writers/CsvBlocksInfoWriter.scala

```diff
@@ -0,0 +1,29 @@
+package beam.utils.scenario.generic.writers
+
+import beam.utils.csv.CsvWriter
+import beam.utils.scenario.{BlockInfo, HouseholdInfo}
+
+import scala.util.Try
+
+trait CsvBlocksInfoWriter {
+  def write(path: String, xs: Iterable[BlockInfo]): Unit
+}
+
+object CsvBlocksInfoWriter extends CsvBlocksInfoWriter {
+  private val headers: Array[String] = Array("blockId", "x", "y")
+
+  override def write(path: String, xs: Iterable[BlockInfo]): Unit = {
+    val csvWriter = new CsvWriter(path, headers)
+    try {
+      xs.foreach { household =>
+        csvWriter.write(
+          household.blockId.id,
+          household.x,
+          household.y
+        )
+      }
+    } finally {
+      Try(csvWriter.close())
```

Comment:
Why wrap this in a `Try`. If it is a non-fatal event then we should at least log a failure, and I see no point in using up an object if it isn't being passed back - so a regular try/catch should work.

Reply:
Apparently I copied it from other csv writers. We may want to change them all as following. Let me know what you think. Thanks.

```diff
diff --git a/src/main/scala/beam/utils/csv/CsvWriter.scala b/src/main/scala/beam/utils/csv/CsvWriter.scala
index ddc02b9a7..9e91ca0da 100644
--- a/src/main/scala/beam/utils/csv/CsvWriter.scala
+++ b/src/main/scala/beam/utils/csv/CsvWriter.scala
@@ -1,7 +1,10 @@
 package beam.utils.csv
 
+import com.typesafe.scalalogging.LazyLogging
+
 import java.io.Writer
 import org.matsim.core.utils.io.IOUtils
+
 import scala.util.Try
 
 case class Delimiter(value: String = ",")
@@ -12,7 +15,8 @@ class CsvWriter(
   headers: Seq[String],
   implicit val delimiter: Delimiter = Delimiter(),
   implicit val lineSeparator: LineSeparator = LineSeparator()
-) extends AutoCloseable {
+) extends AutoCloseable
+    with LazyLogging {
   def this(path: String, headers: String*) = this(path, headers)
 
   implicit val writer: Writer = IOUtils.getBufferedWriter(path)
@@ -42,9 +46,10 @@ class CsvWriter(
     CsvWriter.writeLineSeparator
   }
 
-  override def close(): Unit = {
-    Try(writer.close())
-  }
+  override def close(): Unit =
+    Try(writer.close()).failed.foreach { throwable =>
+      logger.error(s"Error while closing csv writer", throwable)
+    }
 
   def writeAllAndClose(rows: Iterable[Seq[Any]]): Try[Unit] = {
     val result = Try(rows.foreach(writeRow))
diff --git a/src/main/scala/beam/utils/scenario/generic/writers/CsvBlocksInfoWriter.scala b/src/main/scala/beam/utils/scenario/generic/writers/CsvBlocksInfoWriter.scala
index e06e30dab..8ffb11412 100644
--- a/src/main/scala/beam/utils/scenario/generic/writers/CsvBlocksInfoWriter.scala
+++ b/src/main/scala/beam/utils/scenario/generic/writers/CsvBlocksInfoWriter.scala
@@ -23,7 +23,7 @@ object CsvBlocksInfoWriter extends CsvBlocksInfoWriter {
         )
       }
     } finally {
-      Try(csvWriter.close())
+      csvWriter.close()
     }
   }
 }

diff --git a/src/main/scala/beam/utils/scenario/generic/writers/CsvParkingInfoWriter.scala b/src/main/scala/beam/utils/scenario/generic/writers/CsvParkingInfoWriter.scala
index 2bb37ca51..73c9656e3 100644
--- a/src/main/scala/beam/utils/scenario/generic/writers/CsvParkingInfoWriter.scala
+++ b/src/main/scala/beam/utils/scenario/generic/writers/CsvParkingInfoWriter.scala
@@ -47,7 +47,7 @@ object CsvParkingInfoWriter extends ParkingInfoWriter {
         }
       }
     } finally {
-      Try(csvWriter.close())
+      csvWriter.close()
     }
   }
 
diff --git a/src/main/scala/beam/utils/scenario/generic/writers/HouseholdInfoWriter.scala b/src/main/scala/beam/utils/scenario/generic/writers/HouseholdInfoWriter.scala
index 11b041b47..6454254e4 100644
--- a/src/main/scala/beam/utils/scenario/generic/writers/HouseholdInfoWriter.scala
+++ b/src/main/scala/beam/utils/scenario/generic/writers/HouseholdInfoWriter.scala
@@ -25,7 +25,7 @@ object CsvHouseholdInfoWriter extends HouseholdInfoWriter {
         )
       }
     } finally {
-      Try(csvWriter.close())
+      csvWriter.close()
     }
   }
 }
diff --git a/src/main/scala/beam/utils/scenario/generic/writers/PersonInfoWriter.scala b/src/main/scala/beam/utils/scenario/generic/writers/PersonInfoWriter.scala
index b8a1d8c25..76a99dfa2 100644
--- a/src/main/scala/beam/utils/scenario/generic/writers/PersonInfoWriter.scala
+++ b/src/main/scala/beam/utils/scenario/generic/writers/PersonInfoWriter.scala
@@ -28,7 +28,7 @@ object CsvPersonInfoWriter extends PersonInfoWriter {
         )
       }
     } finally {
-      Try(csvWriter.close())
+      csvWriter.close()
     }
   }
 }
diff --git a/src/main/scala/beam/utils/scenario/generic/writers/PlanElementWriter.scala b/src/main/scala/beam/utils/scenario/generic/writers/PlanElementWriter.scala
index d5273b7d7..32617b844 100644
--- a/src/main/scala/beam/utils/scenario/generic/writers/PlanElementWriter.scala
+++ b/src/main/scala/beam/utils/scenario/generic/writers/PlanElementWriter.scala
@@ -65,7 +65,7 @@ object CsvPlanElementWriter extends PlanElementWriter {
         )
       }
     } finally {
-      Try(csvWriter.close())
+      csvWriter.close()
     }
 
   }
```

Reply:
That looks good, except I would just use a regular `try`/`catch`. A `Try` object is more for when you need to pass it out of the function. Also we should avoid overly defensive code such as this anyway.